### PR TITLE
fix istioctl analyze report IST0145 error with wildcard host and specific subdomain

### LINKED ISSUE #52413: istioctl analyze shouldn't report IST0145 for wildcard host and specific subdomain
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

istioctl analyze reports IST0145 for gateways when one contains a wildcard host (e.g. *.example.com) and the other a specific subdomain (e.g. foo.example.com). The config works just fine as expected though.

Example config
```
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: example-wildcard
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
    - hosts:
        - '*.example.com'
      port:
        name: example-wildcard
        number: 80
        protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: example-subdomain
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
    - hosts:
        - 'foo.example.com'
      port:
        name: example-subdomain
        number: 80
        protocol: HTTP
```

istioctl analyze output
```
Error [IST0145] (Gateway istio-system/example-subdomain) Conflict with gateways istio-system/example-wildcard (workload selector istio=ingressgateway, port 80, hosts foo.example.com).
Error [IST0145] (Gateway istio-system/example-wildcard) Conflict with gateways istio-system/example-subdomain (workload selector istio=ingressgateway, port 80, hosts *.example.com).
```

### Version

```Text
client version: 1.22.2
control plane version: 1.21.2
data plane version: 1.21.2 (94 proxies)
```

### Additional Information

_No response_

### Affected product area

- [ ] Ambient
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### PR DESCRIPTION
**Pull request by nicole-lihui**
**Please provide a description of this PR:**
resolved #52413

### CODE REVIEW

File: pilot/pkg/model/gateway.go [hunk 1]
```diff
@@ -465,9 +465,9 @@ func GetSNIHostsForServer(server *networking.Server) []string {
 	return sets.SortedList(sniHosts)
 }
 
-// CheckDuplicates returns all of the hosts provided that are already known
+// CheckGatewayHostsDuplicates returns all of the hosts provided that are already known
 // If there were no duplicates, all hosts are added to the known hosts.
-func CheckDuplicates(hosts []string, bind string, knownHosts map[string]string) []string {
+func CheckGatewayHostsDuplicates(hosts []string, bind string, knownHosts map[string]string) []string {
```
Comment (hanxiaop) [inline pilot/pkg/model/gateway.go:470]: Looks like there's no need to rename this function?

Comment (nicole-lihui) [reply pilot/pkg/model/gateway.go:470]: This change does not make much sense, mainly for the convenience of review，two considerations:
1. function names are more readable and easy to reference
2. the code is referenced here for easy review

Comment (nicole-lihui) [reply pilot/pkg/model/gateway.go]: the naming is verbose， It's better to stay the same 😆

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**